orchestrator/tasks.py

```diff
@@ -73,103 +55,169 @@ def on_failure_to_submit(self, exc, task_id, args, kwargs, einfo):
+        job.update_status(Status.RUNNING)
+        return
+    logger.info(
+        "Can't resume job: %s because it is in status %s, not in SUSPENDED" % (Status(job.status).name, str(job.id)))
 
 
 @shared_task
 @memcache_lock("rb_submit_pending_jobs")
-def submit_pending_jobs():
-    jobs_running = len(Job.objects.filter(status__in=(Status.RUNNING, Status.PENDING)))
+def process_jobs():
+    status_jobs = Job.objects.filter(
+        status__in=(Status.SUBMITTED, Status.PENDING, Status.RUNNING, Status.UNKNOWN,)).values_list('pk', flat=True)
+    for job_id in status_jobs:
+        # Send CHECK_STATUS commands for Jobs
+        command_processor.delay(Command(CommandType.CHECK_STATUS_ON_LSF, str(job_id)).to_dict())
+
+    jobs_running = Job.objects.filter(
+        status__in=(Status.SUBMITTING, Status.SUBMITTED, Status.PENDING, Status.RUNNING,)).count()
     jobs_to_submit = MAX_RUNNING_JOBS - jobs_running
     if jobs_to_submit <= 0:
         return
 
-    job_ids = Job.objects.filter(status=Status.CREATED).order_by("created_date").values_list('pk', flat=True)[
-              :jobs_to_submit]
+    jobs = Job.objects.filter(status=Status.CREATED).order_by("created_date")[
+           :jobs_to_submit]
 
-    Job.objects.filter(pk__in=list(job_ids)).update(status=Status.PENDING)
+    for job in jobs:
+        # Send SUBMIT commands for Jobs
+        with transaction.atomic():
+            if Status(job.status).transition(Status.SUBMITTING):
+                job.update_status(Status.SUBMITTING)
+                command_processor.delay(Command(CommandType.SUBMIT, str(job.id)).to_dict())
 
-    for job_id in job_ids:
-        submit_job_to_lsf.delay(job_id)
+
+@shared_task(bind=True)
+def command_processor(self, command_dict):
+    try:
+        command = Command.from_dict(command_dict)
+        lock_id = "job_lock_%s" % command.job_id
+        with memcache_task_lock(lock_id, self.app.oid) as acquired:
+            if acquired:
+                try:
+                    job = Job.objects.get(id=command.job_id)
+                except Job.DoesNotExist:
+                    return
+                if command.command_type == CommandType.SUBMIT:
+                    submit_job_to_lsf(job)
+                elif command.command_type == CommandType.CHECK_STATUS_ON_LSF:
+                    check_job_status(job)
+                elif command.command_type == CommandType.ABORT:
+                    abort_job(job)
+                elif command.command_type == CommandType.SUSPEND:
+                    suspend_job(job)
+                elif command.command_type == CommandType.RESUME:
+                    resume_job(job)
+            else:
+                self.retry()
+    except RetryException as e:
+        logger.warning(
+            "Command %s failed. Retrying in %s. Excaption %s" % (command_dict, self.request.retries * 5, str(e)))
+        raise self.retry(exc=e, countdown=self.request.retries * 5, max_retries=5)
```

Comment:
I think the retry should last as long as the memcache timeout. So maybe isntead we do something exponential.
If we set max_retries to 6 and countdown to `self.request.retries`, 3 ** 6 will be slightly over 10 min